clang/test/SemaSYCL/accessor-type-diagnostics.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -I %S/Inputs -fsycl -triple spir64 -fsycl-is-device -verify -fsyntax-only  %s
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -triple spir64 -fsycl-is-device -verify -fsyntax-only -Wno-sycl-2017-compat  %s
```

Comment:
Why do need to add `-Wno-sycl-2017-compat` to so many tests?

Reply:
We are moving the ersatz kernels and functor declarations to be compatible with SYCL 2020. But right now, SYCL 1.2.1 is still the default choice in the front end - this mismatch generates a warning, which we suppress with the flag.

Reply:
Considering how many SYCL-2020 features are enabled by default in our implementation, it might be reasonable to set SYCL-2020 by default (although the implementation is not complete). What do you think?

Reply:
It's an easy change to make and requires only two minor updates to the lit tests.  
But would there be any impact on the running of the SYCL conformance tests?  This PR has already drifted a bit too much for my taste. Let's make switching to 2020 as the default it's own work item with its own PR.

Reply:
Okay.